### PR TITLE
Relax version requirements for dependencies

### PR DESCRIPTION
**Pull request by uint**
@webmaster128 Can you think of any reason `cw-storage-plus` together with `cosmwasm-std 1.0.0` could be problematic? I tried it, tests pass, all seems good.

### CODE REVIEW

**Comment by webmaster128** [comment]
I like! Can you check out https://twitter.com/jonhoo/status/1571290371124260865 and related discussions, especially around `-Zminimal-versions`? I never tried that but might be word another CI job here.

**Comment by uint** [comment]
> I like! Can you check out https://twitter.com/jonhoo/status/1571290371124260865 and related discussions, especially around -Zminimal-versions? I never tried that but might be word another CI job here.

Yeah, that sounds like a really good idea. Let's try it. I wonder if we could have the minimal/maximal build be nightly in the future.

I initially removed the lockfile from this repo to follow "Rust best practices", but this convinces me we want the lockfile checked in. Going to revert that.

**Comment by uint** [comment]
@webmaster128 Hah. Currently `cargo build -Zminimal-versions` fails for `cosmwasm-crypto`. I think we'd have to start there.

**Comment by uint** [comment]
@webmaster128 How's this?

**Comment by webmaster128** [comment]
What's the plan to get `-Zminimal-versions` to work? I can run `cargo +nightly build -Zminimal-versions` in `cosmwasm/packages/crypto` without issues. Is there some trick?

**Comment by uint** [comment]
> What's the plan to get -Zminimal-versions to work? I can run cargo +nightly build -Zminimal-versions in cosmwasm/packages/crypto without issues. Is there some trick?

Can you try deleting the lockfile before you run it? In my runs, it seemed like it would only pull minimal dependencies that way.

**Comment by webmaster128** [comment]
If we wait for https://github.com/CosmWasm/cosmwasm/pull/1465 we can finalize it here. Or do you have a need for this PR to be merged?

**Comment by webmaster128** [comment]
> If we wait for [CosmWasm/cosmwasm#1465](https://github.com/CosmWasm/cosmwasm/pull/1465) we can finalize it here. Or do you have a need for this PR to be merged?

Nevermind, that's a bad idea. Then you had to set the min cosmwasm-std version to the latest patch release just because some dependencies further down the stack.

File: .circleci/config.yml [hunk 1]
```diff
@@ -56,21 +66,73 @@ jobs:
...
+          key: cargocache-v2-storage-plus:1.64.0-minimal-{{ checksum "~/project/Cargo.toml" }}
+
+  build_maximal:
     docker:
       - image: rust:1.64.0
+    working_directory: ~/project/
     steps:
-      - checkout
+      - checkout:
+          path: ~/project
```
Comment (webmaster128) [inline .circleci/config.yml:103]: Isn't that the default folder?

File: .circleci/config.yml [hunk 2]
```diff
@@ -56,21 +66,73 @@ jobs:
...
+          name: Build library for native target (with iterator and macro)
+          command: cargo build --locked --all-features
+      - run:
+          name: Run unit tests (with iterator and macro)
+          command: cargo test --workspace --locked --all-features
+      - save_cache:
+          paths:
+            - /usr/local/cargo/registry
+            - target
+          key: cargocache-v2-storage-plus:1.64.0-{{ checksum "~/project/Cargo.lock" }}
```
Comment (webmaster128) [inline .circleci/config.yml:123]: We just use `{{ checksum "Cargo.lock" }}` in cosmwasm

**Review by webmaster128 — APPROVED** [review]
Looks good. Just a bit confusion around the default folder.

**Comment by uint** [comment]
> Nevermind, that's a bad idea. Then you had to set the min cosmwasm-std version to the latest patch release just because some dependencies further down the stack.

I wonder if I could set those further dependencies explicitly here as a workaround, just to make `-Zminimal-versions` work even with `cosmwasm-std 1.0.0`. Is that too much of a hack?

File: .circleci/config.yml (hunk 1 above)
Comment (uint) [reply .circleci/config.yml:103]: I'm sure it is. I copy-pasted this from somewhere without thinking. Thanks!

File: .circleci/config.yml (hunk 2 above)
Comment (uint) [reply .circleci/config.yml:123]: Sounds good

**Comment by webmaster128** [comment]
> I wonder if I could set those further dependencies explicitly here as a workaround, just to make `-Zminimal-versions` work even with `cosmwasm-std 1.0.0`. Is that too much of a hack?

It can be done the same way as in the cosmwasm PR. I think as long as we have some comments around it, this is fine. But make sure to have separate PRs for those to ensure you don't introduce dependencies that you don't need otherwise.

**Comment by uint** [comment]
> But make sure to have separate ~PRs~ commits for those to ensure you don't introduce dependencies that you don't need otherwise.

How do you mean? I made sure to rebuild after adding every dep and seeing if that fixed an error for me, but I didn't bother with separate commits per every dep.

**Comment by webmaster128** [comment]
Basically I wanted you to avoid adding the wasmer and clap stuff that we only need for cosmwasm-vm, not cosmwasm-std. Commit looks great 🏅

**Comment by uint** [comment]
> Basically I wanted you to avoid adding the wasmer and clap stuff that we only need for cosmwasm-vm, not cosmwasm-std. 

Ahhh, I see. I get the gist - I'll avoid adding all those deps by bulk :+1:

**Comment by uint** [comment]
I wonder if we want the minimal/maximal builds to be visible on PRs after all. Someone could change the deps and break those builds, and we want to know in the PR, not later.

**Pull request merged**